src/Surfnet/StepupMiddleware/ApiBundle/Controller/DeprovisionController.php

```diff
@@ -18,20 +18,46 @@
 
 namespace Surfnet\StepupMiddleware\ApiBundle\Controller;
 
+use Surfnet\StepupMiddleware\ApiBundle\Service\DeprovisionServiceInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\JsonResponse;
 
 class DeprovisionController extends AbstractController
 {
-    public function deprovisionAction()
+    private $deprovisionService;
+
+    private $applicationName;
+
+    public function __construct(DeprovisionServiceInterface $deprovisionService, string $applicationName)
+    {
+        $this->deprovisionService = $deprovisionService;
+        $this->applicationName = $applicationName;
+    }
+
+    public function deprovisionAction(string $collabPersonId): JsonResponse
     {
         $this->denyAccessUnlessGranted(['ROLE_DEPROVISION']);
-        return new JsonResponse();
+
+        $userData = $this->deprovisionService->readUserData($collabPersonId);
+        if (!empty($userData)) {
+            $this->deprovisionService->deprovision($collabPersonId);
+        }
+        return new JsonResponse($this->formatResponse('OK', $userData));
```

Comment:
Should this return OK when a user is not found by the DeprovisioningService?

Reply:
Yes, good point! I will ensure that other status responses will not return a 200, OK response.

For now I'm leaving this as it is.